DOCSUP-8939: document the insert_null_as_default setting (#24430)
ClickHouse/ClickHouse: docs/en/operations/settings/settings.md

## input_format_null_as_default {#settings-input-format-null-as-default}

> Enables or disables insertion of default values if input data contain `NULL`, but the data type of the corresponding column in not `Nullable(T)` (for text input formats). This setting is applied to `INSERT ... VALUES` queries.

Reviewer: ```suggestion
For text input formats initialize `NULL` fields with default values if data type of this field is not nullable.
```

Author: I mentioned `INSERT ... VALUES` query to clear the difference between **input_format_null_as_default** and **insert_null_as_default** settings.
Can we leave your variant and append _This setting is applied (or applicable) to INSERT ... VALUES queries_ to it?

Author: What does _For text input formats_ mean?
It seems that this setting is also used for numeric data types. I have checked Int32:
```
CREATE TABLE test_null_as_default (a String DEFAULT '_', b Int32 DEFAULT 99) ENGINE = Memory;

SET input_format_null_as_default=1;
INSERT INTO test_null_as_default (a, b) VALUES ('A0', 10), (NULL, NULL);
SELECT * FROM test_null_as_default;
```
```
A0      10
_       99
```

Reviewer: > Can we leave your variant and append

yes, ok.

> What does For text input formats mean?

It means CSV, TSV, etc.

## insert_null_as_default {#insert_null_as_default}

> Enables the insertion of [default values](../../sql-reference/statements/create/table.md#create-default-values) instead of [NULL](../../sql-reference/syntax.md#null-literal) into non-[Nullable](../../sql-reference/data-types/nullable.md#data_type-nullable) columns. This setting is applied to `INSERT ... SELECT` [queries](../../sql-reference/statements/insert-into.md#insert_query_insert-select). `SELECT` subqueries may be concatenated with `UNION ALL` clause.

Reviewer: ```suggestion
Enables the insertion of [default values](../../sql-reference/statements/create/table.md#create-default-values) instead of [NULL](../../sql-reference/syntax.md#null-literal) into non-[Nullable](../../sql-reference/data-types/nullable.md#data_type-nullable) columns. This setting is applicable to `INSERT ... SELECT` [queries](../../sql-reference/statements/insert-into.md#insert_query_insert-select). `SELECT` subqueries may be concatenated with `UNION ALL` clause.
```

If column type is Nullable then `NULL` values are inserted as is.

> If column type is non-Nullable and this setting is disabled then inserting `NULL` causes an exception.

Reviewer: ```suggestion
If column type is non-Nullable and this setting is disabled, then inserting `NULL` causes an exception.
```

## input_format_null_as_default {#settings-input-format-null-as-default}

> For text input formats initializes `NULL` fields with default values if data type of these fields is not nullable.
> This setting is applied to `INSERT ... VALUES` queries.

Reviewer 2: ```suggestion
Enables or disables using default values if input data contains `NULL`, but the data type of the corresponding column in not `Nullable(T)` (for text input formats). 
This setting is applied to [INSERT ... VALUES](../../sql-reference/statements/insert-into.md) queries.
```

> This setting is applicable to `INSERT ... VALUES` queries for text input formats.

Reviewer 2: ```suggestion
This setting is applicable to [INSERT ... VALUES](../../sql-reference/statements/insert-into.md) queries for text input formats.
```

## insert_null_as_default {#insert_null_as_default}

> Enables the insertion of [default values](../../sql-reference/statements/create/table.md#create-default-values) instead of [NULL](../../sql-reference/syntax.md#null-literal) into clumns with not [nullable](../../sql-reference/data-types/nullable.md#data_type-nullable) data type. 

Reviewer 2: ```suggestion
Enables the insertion of [default values](../../sql-reference/statements/create/table.md#create-default-values) instead of [NULL](../../sql-reference/syntax.md#null-literal) into columns with not [nullable](../../sql-reference/data-types/nullable.md#data_type-nullable) data type. 
```

> This setting is applied to `INSERT ... SELECT` [queries](../../sql-reference/statements/insert-into.md#insert_query_insert-select). Note that `SELECT` subqueries may be concatenated with `UNION ALL` clause.

Reviewer 2: ```suggestion
This setting is applied to [INSERT ... SELECT](../../sql-reference/statements/insert-into.md#insert_query_insert-select) queries. Note that `SELECT` subqueries may be concatenated with `UNION ALL` clause.
```

> If column type is nullable then `NULL` values are inserted as is regardless of this setting.

Reviewer 2: ```suggestion
If column type is nullable then `NULL` values are inserted as is, regardless of this setting.
```
